python/tvm/micro/device/base.py

```diff
@@ -92,71 +95,142 @@ def create_micro_lib_base(
 
     options : List[str]
         additional options to pass to GCC
+
+    lib_src_paths : Optional[List[str]]
+        paths to additional source files to be compiled into the library
     """
+    # look at these (specifically `strip`):
+    #   https://stackoverflow.com/questions/15314581/g-compiler-flag-to-minimize-binary-size
     base_compile_cmd = [
-        f"{toolchain_prefix}gcc",
-        "-std=c11",
-        "-Wall",
-        "-Wextra",
-        "--pedantic",
-        "-c",
-        "-O0",
-        "-g",
-        "-nostartfiles",
-        "-nodefaultlibs",
-        "-nostdlib",
-        "-fdata-sections",
-        "-ffunction-sections",
+        f'{toolchain_prefix}gcc',
```

Comment:
undid all the quote conversions, except those that made sense (i.e. to avoid escapes, as in f'unknown variable "{var}"'). can you point out any other changes you want me to revert? I didn't originally author most of this code so I don't have all the context in my head.